src/saltext/azurerm/utils/azurerm.py

```diff
@@ -103,12 +104,15 @@ def _determine_auth(**kwargs):
                 "populated if using username/password authentication."
             )
         else:
-            credentials = UserPassCredentials(
-                kwargs["username"], kwargs["password"], cloud_environment=cloud_env
+            credentials = UsernamePasswordCredential(
+                client_id=kwargs["client_id"],
+                username=kwargs["username"],
+                password=kwargs["password"],
+                cloud_environment=cloud_env,
             )
     elif "subscription_id" in kwargs:
         try:
-            credentials = MSIAuthentication(cloud_environment=cloud_env)
+            credentials = DefaultAzureCredential(cloud_environment=cloud_env)
```

Comment:
**chore (blocking):** Let's find someplace to document the different types of auth we support now that we're using azure-identity. We could put it in a docstring somewhere and maybe add a static doc location about identity.